socs/agents/hwp_pcu/agent.py

```diff
@@ -0,0 +1,245 @@
+            if not acquired:
+                self.log.warn(
+                    'Could not run init_connection because {} is already running'.format(self.lock.job))
+                return False, 'Could not acquire lock'
+
+            try:
+                self.PCU = pcu.PCU(port=self.port)
+                self.log.info('Connected to PCU')
+            except BrokenPipeError:
+                self.log.error('Could not establish connection to PCU')
+                reactor.callFromThread(reactor.stop)
+                return False, 'Unable to connect to PCU'
+        self.status = self.PCU.get_status()
+        self.initialized = True
+
+        # Start 'acq' Process if requested
+        if params['auto_acquire']:
+            self.agent.start('acq')
+
+        return True, 'Connection to PCU established'
+
+    @ocs_agent.param('command', default='off', type=str)
+    def send_command(self, session, params):
+        """send_command(command)
+
+        **Task** - Send commands to the phase compensation unit.
+        off: The compensation phase is zero.
+        on_1:The compensation phase is +120 deg.
+        on_2: The compensation phase is -120 deg.
+        hold: Stop the HWP spin.
+
+        Parameters:
+            command (str): set the operation mode from 'off', 'on_1', 'on_2' or 'hold'.
+
+        """
+        with self.lock.acquire_timeout(3, job='send_command') as acquired:
+            if not acquired:
+                self.log.warn('Could not send command because {} is already running'.format(self.lock.job))
+                return False, 'Could not acquire lock'
+
+        command = params['command']
+        if command == 'off':
+            off_channel = [0, 1, 2, 5, 6, 7]
+            for i in off_channel:
+                self.PCU.relay_off(i)
+            self.status = 'off'
+            msg = 'Phase compensation is "off".'
+            return True, msg
+
+        elif command == 'on_1':
+            on_channel = [0, 1, 2]
+            off_channel = [5, 6, 7]
+            for i in on_channel:
+                self.PCU.relay_on(i)
+            for i in off_channel:
+                self.PCU.relay_off(i)
+            self.status = 'on_1'
+            msg = 'Phase compensation operates "on_1".'
+            return True, msg
+
+        elif command == 'on_2':
+            on_channel = [0, 1, 2, 5, 6, 7]
+            for i in on_channel:
+                self.PCU.relay_on(i)
+            self.status = 'on_2'
+            msg = 'Phase compensation operates "on_2".'
+            return True, msg
+
+        elif command == 'hold':
+            on_channel = [0, 1, 2, 5]
+            off_channel = [6, 7]
+            for i in on_channel:
+                self.PCU.relay_on(i)
+            for i in off_channel:
+                self.PCU.relay_off(i)
+            self.status = 'hold'
+            msg = 'Phase compensation operates "hold".'
+            return True, msg
```

Comment:
All of this must be indented one layer so that it is in the lock acquisition block